test/Translate/Ops/binop.p4

```diff
@@ -2,205 +2,393 @@
-// CHECK:         %[[VAL_98:.*]] = p4hir.binop(mul, %[[VAL_96]], %[[VAL_97]]) : !i10i
-// CHECK:         %[[VAL_99:.*]] = p4hir.variable ["r6", init] : <!i10i>
-// CHECK:         p4hir.assign %[[VAL_98]], %[[VAL_99]] : <!i10i>
-// CHECK:         %[[VAL_100:.*]] = p4hir.read %[[VAL_72]] : <!i10i>
-// CHECK:         %[[VAL_101:.*]] = p4hir.read %[[VAL_75]] : <!i10i>
-// CHECK:         %[[VAL_102:.*]] = p4hir.binop(sadd, %[[VAL_100]], %[[VAL_101]]) : !i10i
-// CHECK:         %[[VAL_103:.*]] = p4hir.variable ["r7", init] : <!i10i>
-// CHECK:         p4hir.assign %[[VAL_102]], %[[VAL_103]] : <!i10i>
-// CHECK:         %[[VAL_104:.*]] = p4hir.read %[[VAL_72]] : <!i10i>
-// CHECK:         %[[VAL_105:.*]] = p4hir.read %[[VAL_75]] : <!i10i>
-// CHECK:         %[[VAL_106:.*]] = p4hir.binop(ssub, %[[VAL_104]], %[[VAL_105]]) : !i10i
-// CHECK:         %[[VAL_107:.*]] = p4hir.variable ["r8", init] : <!i10i>
-// CHECK:         p4hir.assign %[[VAL_106]], %[[VAL_107]] : <!i10i>
-// CHECK:         %[[VAL_108:.*]] = p4hir.read %[[VAL_72]] : <!i10i>
-// CHECK:         %[[VAL_109:.*]] = p4hir.read %[[VAL_75]] : <!i10i>
-// CHECK:         %[[VAL_110:.*]] = p4hir.binop(or, %[[VAL_108]], %[[VAL_109]]) : !i10i
-// CHECK:         %[[VAL_111:.*]] = p4hir.variable ["r11", init] : <!i10i>
-// CHECK:         p4hir.assign %[[VAL_110]], %[[VAL_111]] : <!i10i>
-// CHECK:         %[[VAL_112:.*]] = p4hir.read %[[VAL_72]] : <!i10i>
-// CHECK:         %[[VAL_113:.*]] = p4hir.read %[[VAL_75]] : <!i10i>
-// CHECK:         %[[VAL_114:.*]] = p4hir.binop(and, %[[VAL_112]], %[[VAL_113]]) : !i10i
-// CHECK:         %[[VAL_115:.*]] = p4hir.variable ["r12", init] : <!i10i>
-// CHECK:         p4hir.assign %[[VAL_114]], %[[VAL_115]] : <!i10i>
-// CHECK:         %[[VAL_116:.*]] = p4hir.read %[[VAL_72]] : <!i10i>
-// CHECK:         %[[VAL_117:.*]] = p4hir.read %[[VAL_75]] : <!i10i>
-// CHECK:         %[[VAL_118:.*]] = p4hir.binop(xor, %[[VAL_116]], %[[VAL_117]]) : !i10i
-// CHECK:         %[[VAL_119:.*]] = p4hir.variable ["r13", init] : <!i10i>
-// CHECK:         p4hir.assign %[[VAL_118]], %[[VAL_119]] : <!i10i>
+// CHECK: #[[const_int2_b10i:.+]] = #p4hir.int<2> : !b10i
+// CHECK: #[[const_int2_i10i:.+]] = #p4hir.int<2> : !i10i
+// CHECK: #[[const_int3_b10i:.+]] = #p4hir.int<3> : !b10i
+// CHECK: #[[const_int3_i10i:.+]] = #p4hir.int<3> : !i10i
+
+// CHECK-LABEL:   p4hir.func action @bit_add_bit
+// CHECK:           %[[var_lhs:.*]] = p4hir.variable ["lhs"] : <!b10i>
+// CHECK:           %[[var_rhs:.*]] = p4hir.variable ["rhs"] : <!b10i>
+// CHECK:           %[[val_lhs:.*]] = p4hir.read %[[var_lhs]] : <!b10i>
+// CHECK:           %[[val_rhs:.*]] = p4hir.read %[[var_rhs]] : <!b10i>
+// CHECK:           %[[val_binop_add:.*]] = p4hir.binop(add, %[[val_lhs]], %[[val_rhs]]) : !b10i
+// CHECK:           %[[var_res:.*]] = p4hir.variable ["res", init] : <!b10i>
+// CHECK:           p4hir.assign %[[val_binop_add]], %[[var_res]] : <!b10i>
+// CHECK:           p4hir.return
+
+action bit_add_bit() {
```

Comment:
To simplify things here and elsewhere: maybe we can just have `lhs` and `rhs` as `in` arguments? and `res` as an `out` argumebnt?